### PR TITLE
feat: generate host name based on a random number

### PR DESCRIPTION
**Pull request by Iceber**
## Feature or Problem

Generate a friendly name for the host based on a random number.
Names we pulled from a list of friendly or neutral adjectives and nouns suitable for use in public and on hosts/domain names

bring the adjectives and nouns over from https://github.com/wasmCloud/wasmcloud-otp/blob/main/host_core/lib/host_core/namegen.ex
## Related Issues

fix: https://github.com/wasmCloud/wasmCloud/issues/491
## Testing

Built on platform(s)
- [ ] x86_64-linux
- [ ] aarch64-linux
- [ ] x86_64-darwin
- [x] aarch64-darwin
- [ ] x86_64-windows

Tested on platform(s)
- [ ] x86_64-linux
- [ ] aarch64-linux
- [ ] x86_64-darwin
- [ ] aarch64-darwin
- [ ] x86_64-windows

### CODE REVIEW

File: crates/host/src/wasmbus/mod.rs [hunk 1]
```diff
@@ -895,6 +895,38 @@ async fn create_lattice_metadata_bucket(
...
+    haze mountain night pond darkness snowflake silence
+    sound sky shape stapler surf thunder violet water wildflower
+    wave water resonance sun wood dream cherry tree fog autocorrect
+    frost voice paper frog smoke star hamster ocean emoji robot
+    ";
+
+    /// Generate a friendly name for the host based on a random number.
+    /// Names we pulled from a list of friendly or neutral adjectives and nouns suitable for use in
+    /// public and on hosts/domain names
+    fn generate_friendly_name() -> Option<String> {
```
Comment (Iceber) [inline crates/host/src/wasmbus/mod.rs:924]: I considered putting it in the new names.mod file, but it looks like it's only necessary for use here, so it's in Host.

File: crates/host/src/wasmbus/mod.rs [hunk 2]
```diff
@@ -895,6 +895,38 @@ async fn create_lattice_metadata_bucket(
...
+    const NAME_ADJECTIVES: &str = "
+    autumn hidden bitter misty silent empty dry dark summer
+    icy delicate quiet white cool spring winter patient
+    twilight dawn crimson wispy weathered blue billowing
+    broken cold damp falling frosty green long late lingering
+    bold little morning muddy old red rough still small
+    sparkling bouncing shy wandering withered wild black
+    young holy solitary fragrant aged snowy proud floral
+    restless divine polished ancient purple lively nameless
+    gray orange mauve
```
Comment (Iceber) [inline crates/host/src/wasmbus/mod.rs:907]: I didn't convert it to `&[&str]`, string look better maintained.
```rust
const NAME_ADJECTIVES: &[&str] = [
"autumn",
"hidden",
"bitter",
... // A very long list
]

**Review by rvolosatovs — APPROVED** [review]
Thank you for contribution!

File: crates/host/src/wasmbus/mod.rs (hunk 2 above)
Comment (rvolosatovs) [reply crates/host/src/wasmbus/mod.rs:907]: I would prefer an array here, but I understand the reasoning and we can always optimize later if we need to (we most probably will not)

Comment (Iceber) [reply crates/host/src/wasmbus/mod.rs:907]: I tried the array and it is very long after formatting.:joy:

**Pull request merged**